### PR TITLE
Support using FIRRTL2 as input for Chipyard's version of FireSim

### PR DESCRIPTION
**Pull request by abejgonzalez**
Allows FireSim to consume FIRRTL2 `.fir` files while being able to ingest Chisel6 anno files. Also removes some more APIs in `firesim-lib` and `targetutils` that aren't supported in newer versions of Chisel.

#### Contributor Checklist
- [ ] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [ ] Did you add Scaladoc/docstring/doxygen to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous prints/debugging code?
- [ ] Did you state the UI / API impact?
- [ ] Did you specify the Verilog / AGFI compatibility impact?

- [ ] If applicable, did you regenerate and publicly share default AGFIs?

- [ ] If applicable, did you apply the `Please Backport` label?

#### Reviewer Checklist (only modified by reviewer)
Note: to run CI on PRs from forks, comment `@Mergifyio copy main` and manage the change from the new PR.
- [ ] Is the title suitable for inclusion in the changelog and does the PR have a `changelog:<topic>` label?
- [ ] Did you mark the proper release milestone?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?

#### CI Help
Add the following labels to modify the CI for a set of features.
Generally, a label added only affect subsequent changes to the PR (i.e. new commits, force pushing, closing/reopening).
See `ci:*` for full list of labels:
- `ci:fpga-deploy` - Run FPGA-based E2E testing
- `ci:local-fpga-buildbitstream-deploy` - Build local FPGA bitstreams for platforms that are released
- `ci:persist-prior-workflows` - Prevent prior CI workflows from automatically cancelling with subsequent changes
- `ci:disable` - Disable CI
- `ci:disable-scala-tests` - Disable Scala tests in CI

### CODE REVIEW

File: deploy/util/targetprojectutils.py
Comment (abejgonzalez) [inline deploy/util/targetprojectutils.py:1]: Should be separate PR.

**Comment by joonho3020** [comment]
How are you planning to generate FIRRTL2 in CY?
Otherwise SFTM

**Comment by abejgonzalez** [comment]
> How are you planning to generate FIRRTL2 in CY? Otherwise SFTM

https://github.com/ucb-bar/chipyard/pull/2013

**Review by joonho3020 — APPROVED** [review]
